miso-web/src/spring-test/java/uk/ac/bbsrc/tgac/miso/webapp/springtest/HotRestControllerST.java

```diff
@@ -0,0 +1,118 @@
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.*;
+import org.springframework.test.web.servlet.ResultActions;
+import com.jayway.jsonpath.JsonPath;
+
+import static org.hamcrest.Matchers.*;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+
+import org.springframework.test.web.servlet.MvcResult;
+import uk.ac.bbsrc.tgac.miso.dto.Dtos;
+
+import org.springframework.scheduling.annotation.EnableAsync;
+import org.springframework.web.servlet.View;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import org.springframework.security.test.context.support.WithMockUser;
+import uk.ac.bbsrc.tgac.miso.core.data.type.StatusType;
+import static org.junit.Assert.*;
+import java.util.Collections;
+import uk.ac.bbsrc.tgac.miso.webapp.controller.rest.HotRestController.SpreadsheetDataDto;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.util.List;
+import java.util.Arrays;
+import java.util.ArrayList;
+
+import org.springframework.test.web.servlet.MockMvc;
+import java.util.Date;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import org.springframework.mock.web.MockHttpServletResponse;
+
+
+public class HotRestControllerST extends AbstractST {
+
+  private static final String CONTROLLER_BASE = "/rest/hot";
+
+  @Test
+  public void testDownloadSpreadsheet() throws Exception {
+    SpreadsheetDataDto sheet = new SpreadsheetDataDto();
+
+    ArrayList<String> headers = new ArrayList<String>(Arrays.asList("1", "2", "3", "4", "5"));
+    ArrayList<String> firstrow = new ArrayList<String>(Arrays.asList("one", "two", "three", "four", "five"));
+    ArrayList<String> secondrow = new ArrayList<String>(Arrays.asList("un", "deux", "trois", "quatre", "cinq"));
+    List<List<String>> rows = new ArrayList<List<String>>(Arrays.asList(firstrow, secondrow));
+
+    sheet.setHeaders(headers);
+    sheet.setRows(rows);
+
+    MockHttpServletResponse response = getMockMvc()
+        .perform(post(CONTROLLER_BASE + "/spreadsheet").param("format", "CSV").content(makeJson(sheet))
+            .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(status().isOk())
+        .andExpect(content().contentType("text/csv"))
+        .andExpect(header().longValue("Content-Length", 83)).andReturn().getResponse();
+
+    String filename = response.getHeader("Content-Disposition").split("=")[1];
+    List<List<String>> records = new ArrayList<>();
+    try (BufferedReader br = new BufferedReader(new FileReader(filename))) {
+      String line;
+      int row = 0;
+      while ((line = br.readLine()) != null) {
+        String[] values = line.split(",");
+        records.add(Arrays.asList(values));
+        switch (row) {
+          case 0:
+            assertEquals(values[0], "1");
```

Comment:
Would be more thorough and simpler to write to use loop counters to check the entire file against `rows`